lib/commons/color/get-background-color.js

```diff
@@ -207,17 +207,20 @@ color.getRectStack = function(elm) {
 function sortPageBackground(elmStack) {
 	let bodyIndex = elmStack.indexOf(document.body);
 	let bgNodes = elmStack;
+	let sortBodyElement =
+		bodyIndex > 1 || bodyIndex === -1; // only if there are negative z-index elements // only when an element is positioned outside of the body's rect
 
 	if (
-		// Check that the body background is the page's background
-		bodyIndex > 1 && // only if there are negative z-index elements
+		sortBodyElement &&
 		!color.elementHasImage(document.documentElement) &&
 		color.getOwnBackgroundColor(
 			window.getComputedStyle(document.documentElement)
 		).alpha === 0
 	) {
-		// Remove body and html from it's current place
-		bgNodes.splice(bodyIndex, 1);
+		// Only remove document.body if it was originally contained within the element stack
+		if (bodyIndex !== -1) {
```

Comment:
This looks wrong to me. My understanding is that the body is **always** at the back. Even if it has a height of 1px, and elementsFromPoint doesn't pick it up, the browser will render it over the whole page anyway. It seems to me like you should also do this splice when `bodyIndex === -1` (so remove the if).

Otherwise what you'll get is (like in your test case) a black background, even though the body has a different color.

Reply:
It doesn't look that way, at least according to the [CSS3 spec](https://www.w3.org/TR/css-backgrounds-3/#special-backgrounds):

> For documents whose root element is an HTML HTML element or an XHTML html element [HTML]: if the computed value of background-image on the root element is none and its background-color is transparent, user agents must instead propagate the computed values of the background properties from that element’s first HTML BODY or XHTML body child element. The used values of that BODY element’s background properties are their initial values, and the propagated values are treated as if they were specified on the root element.

The `<html>` or document canvas just inherits the color from whatever the root element is, and doesn't expand the root element itself.

Here's a test case that demonstrates this: https://jsfiddle.net/o3hjdsy8/1/ Even though the body has an implicit height of `10px`, the canvas still fills the screen with the background color according to the spec.

I tested this in:

* Chrome 73
* Firefox 66
* Safari 12

When getting the center of the element with `document.elementsFromPoint(x, y)` the element stack does not include `<body>` in any of the above browsers.